src/transactions/test/TxEnvelopeTests.cpp

```diff
@@ -1022,133 +1022,6 @@ TEST_CASE("txenvelope", "[tx][envelope]")
         }
     }
 
-    SECTION("common transaction")
```

Comment:
those are not duplicate tests so this refactor is incomplete:
they are calling "applyCheck" directly.

I suspect that the problem is that the refactored version calls `validateTxResults` which doesn't check what `apply` returns on failure

Reply:
So do you want to keep tests in TxResultsTests and those here as well?

Reply:
no, it's fine to refactor (as the new approach is more systematic), you just need to make sure that we call `applyCheck` in all cases

Reply:
New version calls `applyCheck` in all cases.

I don't see why in `applyCheck` we need to call `apply` even when `checkValid` failed. This causes many strange edge cases in testing code, as errors reported by `checkValid` and `apply` can be different. I propose that we remove those check and only call `apply` when `checkValid` returned true.

Reply:
replying in the main PR thread